cms/search/tests/test_publishers.py

```diff
@@ -0,0 +1,384 @@
+# test_publishers.py
+import logging
+from datetime import timedelta
+from unittest.mock import ANY, MagicMock, patch
+
+from django.test import TestCase, override_settings
+from django.utils import timezone
+
+from cms.articles.tests.factories import StatisticalArticlePageFactory
+from cms.methodology.tests.factories import MethodologyPageFactory
+from cms.release_calendar.enums import ReleaseStatus
+from cms.release_calendar.tests.factories import ReleaseCalendarPageFactory
+from cms.search.publishers import BasePublisher, KafkaPublisher, LogPublisher
+from cms.standard_pages.tests.factories import IndexPageFactory, InformationPageFactory
+
+EXPECTED_CONTENT_TYPES = {
+    "ReleaseCalendarPage": "release",
+    "StatisticalArticlePage": "bulletin",
+    "InformationPage": "static_page",
+    "IndexPage": "static_landing_page",
+    "MethodologyPage": "static_methodology",
+}
+
+
+class DummyPublisher(BasePublisher):
+    """Concrete subclass of BasePublisher for testing the shared functionality."""
+
+    def __init__(self, channel_created_or_updated, channel_deleted):
+        self._channel_created_or_updated = channel_created_or_updated
+        self._channel_deleted = channel_deleted
+
+    def _publish(self, channel, message):
+        # We don't actually publish in tests; we just want to spy on the calls.
+        pass
+
+    def get_created_or_updated_channel(self):
+        return self._channel_created_or_updated
+
+    def get_deleted_channel(self):
+        return self._channel_deleted
+
+
+class BasePublisherTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.included_factories = [
+            InformationPageFactory,
+            MethodologyPageFactory,
+            ReleaseCalendarPageFactory,
+            StatisticalArticlePageFactory,
+        ]
+
+        cls.index_page = IndexPageFactory(slug="custom-slug-1")
+        cls.included_factories.append(lambda: cls.index_page)
```

Comment:
@sanjeevz3009 please do the same as for https://github.com/ONSdigital/dis-wagtail/pull/115/files/9761f03988ca2eb248cc898fc58cc71307887d72#diff-c7dfda5eebe1d604ad2b46e2ce74f2daa029434e1736d470dd73541a35f9fc78

Reply:
Done!